csrc/multidevice/pipeline.cpp

```diff
@@ -0,0 +1,319 @@
+              stage_desc->mesh.size() == 1,
+              "A global input must belong to a stage which mesh is of size 1");
+        } else {
+          // if the Val is a stage input but not a global input, it must be
+          // defined by a "Set" operation
+          TORCH_INTERNAL_ASSERT(
+              (val->definition()->isA<LoadStoreOp>()) &&
+                  (val->definition()->as<LoadStoreOp>()->opType() ==
+                   LoadStoreOpType::Set),
+              "A Val that is the input of a stage must be defined by a LoadStoreOp expression of type Set"
+              "but here the definition is " +
+                  val->definition()->toString());
+        }
+      }
+      // Create a PipelineVal for each stage's output
+      for (auto val : stage_output_desc_[stage_desc].vector()) {
+        auto p_val =
+            IrBuilder::create<PipelineVal>(pipeline_->as<IrContainer>(), val);
+        val_to_pipeline_val_output_of_stage_[val] = p_val;
+        if (isGlobalOutput(val)) {
+          pipeline_->addOutput(p_val);
+        }
+      }
+    }
+  }
+
+  // Build the PipelineStage IR of the Pipeline
+  // An PipelineStage is created for each Stage of the Fusion
+  void buildPipelineStages() {
+    for (auto& stage_desc : pipeline_->descriptor().stageDescriptors) {
+      // containers for storing the I/O of the PipelineStage
+      ValSet ins, outs;
+      for (auto& val : stage_input_desc_[stage_desc]) {
+        ins.pushBack(val_to_pipeline_val_input_of_stage_[val]);
+      }
+      for (auto& val : stage_output_desc_[stage_desc]) {
+        outs.pushBack(val_to_pipeline_val_output_of_stage_[val]);
+      }
+      auto stage = IrBuilder::create<PipelineStage>(
+          pipeline_->as<IrContainer>(), stage_desc, ins, outs);
+      stages_.push_back(stage);
+    }
+  }
+
+  // Build the PipelineCommunication IR of the Pipeline
+  // A PipelineCommunication is created for each stage's input
+  void buildPipelineCommunications() {
+    for (auto& stage : stages_) {
+      for (auto& p_val : stage->inputs()) {
+        std::vector<Val*> ins, outs;
+        auto val = p_val->as<PipelineVal>()->getOriginalVal();
+        if (isGlobalInput(val)) {
+          continue;
+        }
+        outs.push_back(p_val);
+        for (auto& producer : val->definition()->inputs()) {
+          ins.push_back(val_to_pipeline_val_output_of_stage_.at(producer));
+        }
+        TORCH_INTERNAL_ASSERT(
+            std::size(ins) == 1 && std::size(outs) == 1,
+            "Pipeline Communications must involve one input and one output");
+        IrBuilder::create<PipelineCommunication>(
+            pipeline_->as<IrContainer>(), ins.at(0), outs.at(0));
+      }
+    }
+  }
+};
+
+Pipeline::Pipeline(
+    Fusion* fusion,
+    PipelineDescriptor // NOLINT (pass fusion as value and use std::move)
+        descriptor) // NOLINT (pass fusion as value and use std::move)
+    : original_fusion_(fusion), descriptor_(descriptor) {
```

Comment:
Is the linter asking for the following?
```suggestion
    : original_fusion_(fusion), descriptor_(std::move(descriptor)) {
```

Reply:
that solves the issue, thanks